File: modified_scripts/scala/bubble_sort/deepseek-coder_6.7b.scala

def bubbleSort(xs: Array[Int]): Unit = {
  var swapped = true
  while (swapped) {
    swapped = false
    for (i <- 0 until xs.length - 1) {
      if (xs(i) > xs(i + 1)) {
        val temp = xs(i)
        xs(i) = xs(i + 1)
        xs(i + 1) = temp
        swapped = true
      }
    }
  }
}

File: modified_scripts/r/bubble_sort/deepseek-coder_6.7b.r

bubble_sort <- function(x) {
    n <- length(x)
    repeat {
        swapped <- FALSE
        for (i in 1:(n - 1)) {
            if (x[i] > x[i + 1]) {
                tmp <- x[i]
                x[i] <- x[i + 1]
                x[i + 1] <- tmp
                swapped <- TRUE
            }
        }
        n <- n - 1
        if (!swapped) return(x)
    }
}
